recipes/ruby/all/conanfile.py

```diff
@@ -0,0 +1,192 @@
+import glob
+import os
+import re
+
+from conan import ConanFile
+from conan.tools.apple.apple import is_apple_os, to_apple_arch
+from conan.tools.cross_building import cross_building
+from conan.tools.files import apply_conandata_patches
+from conan.tools.gnu import Autotools, AutotoolsDeps, AutotoolsToolchain
+from conans import tools
+
+required_conan_version = ">=1.43.0"
+
+
+class RubyConan(ConanFile):
+    name = "ruby"
+    description = "The Ruby Programming Language"
+    license = "Ruby"
+    topics = ("ruby", "c", "language", "object-oriented", "ruby-language")
+    homepage = "https://www.ruby-lang.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    exports_sources = "patches/**"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": True
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler in ["Visual Studio", "msvc"]
+
+    @property
+    def _windows_system_libs(self):
+        return ["user32", "advapi32", "shell32", "ws2_32", "iphlpapi", "imagehlp", "shlwapi", "bcrypt"]
+
+    @property
+    def _msvc_runtime_flag(self):        
+        compiler = self.settings.compiler
+        runtime = self.settings.get_safe("compiler.runtime")
+        if compiler == "msvc" or compiler == "intel-cc":
+            build_type = self.settings.get_safe("build_type")
+            if build_type != "Debug":
+                return {"static": "MT",
+                        "dynamic": "MD"}.get(runtime, "")
+            else:
+                return {"static": "MTd",
+                        "dynamic": "MDd"}.get(runtime, "")
+        else:
+            return runtime
+
+    @property
+    def _msvc_optflag(self):
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < "14":
+            return "-O2b2xg-"
+        else:
+            return "-O2sy-"
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+        self.requires("gmp/6.1.2")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1m")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def generate(self):
+        td = AutotoolsDeps(self)
```

Comment:
:wave: @memsharded @franramirez688 Since you both played in the new toolchain... https://github.com/conan-io/conan/pull/10429

Is this ready for use in CCI?